test/unit/finance/bill_variable_for_plan_changed_test.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class Finance::BillVariableForPlanChangedTest < ActiveSupport::TestCase
+  attr_reader :contract, :app_plan
+
+  setup do
+    @contract = FactoryBot.build_stubbed(:contract)
+    account = FactoryBot.build_stubbed(:simple_account)
+    @app_plan = FactoryBot.build_stubbed(:application_plan)
+
+    contract.stubs(:provider_account).returns(account)
+    account.stubs(:provider_can_use?).with(:instant_bill_plan_change).returns(true)
+
+    ENV["TZ"] = "UTC"
+  end
+
+  test "no variable billing on 1st day utc" do
+    Timecop.travel(Time.now.utc.beginning_of_month + 1.day - 1.second)
+
+    contract.expects(:save).never
+    contract.notify_observers(:bill_variable_for_plan_changed, app_plan)
+  end
+
+  test "variable billing after 1st day utc" do
+    Timecop.travel(Time.now.utc.beginning_of_month + 1.day)
+
+    contract.expects(:save).returns(true)
+    contract.notify_observers(:bill_variable_for_plan_changed, app_plan)
+  end
+
+  test "no variable billing on 1st day of month local time" do
+    # this changes server and client local time to China Standard Time (+8)
+    ENV["TZ"] = "Asia/Shanghai"
```

Comment:
Problem is setting `Time.zone = 'Asia/Shanghai'` doesn't change the timezone. Is there a better way to do it?

Reply:
@akostadinov originally added this so maybe he has a better idea why changing the env var

Reply:
But why do you want to work with Timezone? That is the real question.

Reply:
@josemigallas , change the variable and run `Time.now` in pry to see that timezone is changed.

@hallelujah, the test makes sure that billing depends on timezone. If somebody changes `#instant_bill_plan_change` to use `Time.zone.now` instead of the current `Time.now` the test will fail.

Presently no test will fail by such a change.

That is - make sure we retain current behavior until we really want it changed.